ZenPacks/zenoss/LinuxMonitor/parsers/linux/cpu.py

```diff
@@ -35,41 +35,55 @@ def processResults(self, cmd, result):
                    'ssCpuIdlePerCpu',
                    'ssCpuUsedPerCpu',
                    'ssCpuWaitPerCpu']
-            
+
+            # New ids for Device CPU
+            userCpuPerCpu = None
+            systemCpuPerCpu = None
+            idleCpuPerCpu = None
+            usedCpuPerCpu = None
+            waitCpuPerCpu = None
+
+            # Component cpu returns socket value, device cpu return empty str
+            component = getattr(cmd, 'component', None)
+
             # Calculate the number of CPUs
             cpuCount=0
+            single_cpu_line = ''
             for line in cmd.result.output.splitlines()[1:]:
               words = line.split()[0:]
               if 'cpu' in words[0]:
                 cpuCount+=1
+                if component:
+                  if component == line[3]:
+                    # If component, we matched the socket
+                    single_cpu_line = line
+                    break
 
-            # If we got a CPU count, set the normalized perCpu values
-            if cpuCount:
-              perCpuValues = cmd.result.output.splitlines()[0].split()
-              userCpuPerCpu = float(perCpuValues[1])/float(cpuCount)
-              systemCpuPerCpu = float(perCpuValues[3])/float(cpuCount)
-              idleCpuPerCpu = float(perCpuValues[4])/float(cpuCount)
-              usedCpuPerCpu = userCpuPerCpu + systemCpuPerCpu
-              waitCpuPerCpu = float(perCpuValues[5])/float(cpuCount)
+            # Check if component or device cpu monitoting
```

Comment:
This code has a combintation of 2-space and 4-space indentation. Should be consistent.